packages/backends/backend/src/lazy-backend.ts

```diff
@@ -37,8 +44,21 @@ export class LazyBackend implements Backend {
    */
   constructor(private config: BackendConfig) {}
 
+  /**
+   * Initializes the backend instance if it has not already been created.
+   *
+   * If the backend does not exist, this method creates a new backend instance
+   * using the provided configuration. It ensures that the backend is only
+   * initialized once, even if called multiple times concurrently, by caching
+   * the initialization promise.
+   *
+   * @returns {Promise<void>} A promise that resolves when the backend has been initialized.
+   */
   async init(): Promise<void> {
-    this.backend ??= await createBackendFromDriver(this.config);
+    if (!this.backend) {
+      this.backendPromise ??= createBackendFromDriver(this.config);
+      this.backend = await this.backendPromise;
+    }
```

Comment:
If the backendPromise is running and init was called again, `this.backend` is falsy and a new promise is created.

```suggestion
    if (!this.backendPromise) {
        this.backendPromise ??= createBackendFromDriver(this.config);
    } 
    
    this.backend = await this.backendPromise;
```